Add some essential documentation

tadeboro opened the pull request.

Akasurde commented on docs/configuration_management_database.md:
```diff
@@ -0,0 +1,97 @@
+      hosts: localhost
+      gather_facts: false
+
+      tasks:
+	- name: Update the VM state
+	  servicenow.itsm.configuration_item:
```
```suggestion
      tasks:
	  - name: Update the VM state
	    servicenow.itsm.configuration_item:
```

Akasurde commented on docs/general_usage_patterns.md:
```diff
@@ -0,0 +1,213 @@
+	    impact: high
+	    urgency: high
+
+	- name: Delete incident ticket (rare operation)
```
Can you please add these playbooks as separate files and include them in the documentation like https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/scenario_guides/vmware_rest_scenarios/create_vm.rst

tadeboro replied:
Unfortunately, I cannot do this in markdown documents. It would be great if Automation Hub could render rST documentation, but currently, the documentation display is limited to markdown.

mancabizjak commented on docs/general_usage_patterns.md:
```diff
@@ -0,0 +1,213 @@
+	    impact: low
+	    urgency: low
+
+            # Other data
+            other:
+              problem_id: "{{ problem.record.sys_id }}"
+
+	- name: Update an existing incident
+	  servicenow.itsm.incident:
+            # Instance data
+            instance:
+              host: https://dev12345.service-now.com
+              username: user
+              password: pass
+              client_id: cid
+              client_secret: csecret
+
+            # Record identificator
+            number: INC1234567
+
+            # Common parameters
+            state: in_progress
+	    impact: high
+	    urgency: high
+
+	- name: Delete incident ticket (rare operation)
+	  servicenow.itsm.incident:
+            # Instance data
+            instance:
+              host: https://dev12345.service-now.com
+              username: user
+              password: pass
+              client_id: cid
+              client_secret: csecret
+
+            # Record identificator
+            number: INC1234567
+
+            # Common parameters
+            state: absent
```
Inconsistent indentation throughout the entire playbook. I'm looking at the file with "rich diff" option, and it doesn't look the way it's supposed to.

mancabizjak commented on docs/general_usage_patterns.md:
```diff
@@ -0,0 +1,213 @@
+      hosts: localhost
+      gather_facts: false
+
+      tasks:
+	- name: Create a new incident
+	  servicenow.itsm.incident:
+            # No sys_id or number because we are creating a new incident
+
+            # Common parameters
+	    caller: admin
+	    short_description: Demo incident
+	    impact: low
+	    urgency: low
+
+            # Other data
+            other:
+              problem_id: "{{ problem.record.sys_id }}"
+
+	- name: Update an existing incident
+	  servicenow.itsm.incident:
+            # Record identificator
+            number: INC1234567
+
+            # Common parameters
+            state: in_progress
+	    impact: high
+	    urgency: high
+
+	- name: Delete incident ticket (rare operation)
+	  servicenow.itsm.incident:
+            # Record identificator
+            number: INC1234567
+
+            # Common parameters
+            state: absent
```
Please fix inconsistent indentation.

mancabizjak commented on docs/incident_management.md:
```diff
@@ -0,0 +1,174 @@
+fields that we used as task parameters. All other information is retained,
+except for some housekeeping information that ServiceNow updates automatically.
+
+If we run the `ansible-playbook with the `--diff` command-line switch, Ansible
```
```suggestion
If we run the `ansible-playbook` with the `--diff` command-line switch, Ansible
```

mancabizjak reviewed (changes requested):
I noticed a couple of formatting issues (mainly in example playbooks) and inconsistencies. Aside from that, I think the docs are well written and communicate all the relevant info to the end users, so :+1:

tadeboro commented:
I am really sorry for posting badly indented code snippets. I copied the basis from the GitHub repo I used for the demo and failed to notice that GitHub UI treats 8 spaces as one tab character. Ond because my editor is set to render tabs 8 spaces wide, I did not notice this beforehand.

I think `:retab` fixed all the indentation issues.

mancabizjak reviewed (approved):
Thanks @tadeboro, looks good to me!

Akasurde commented:
@tadeboro @mancabizjak I suggested some changes, I would appreciate it if those were included in this PR.

tadeboro commented:
> I suggested some changes, I would appreciate it if those were included in this PR.

I addressed both of your comments: I fixed the indentation in an example you pointed out and I explained why we cannot include examples in our documentation (because AH only knows about markdown and the example you showed uses reStructuredText).

Did you want any other changes to be made?
